### DISCUSSION TITLE
Can't read Jandex index from /META-INF/jandex.idx: Can't read index version 12; this IndexReader only supports index versions 2-3,6-11

### DISCUSSION THREAD

**Discussion by hablocher** [quarkusio/quarkus#41492] [Q&A]
Hi folks! 

How are you? 

I need some help with this error, please!

Let me explain the scenario: 

Everything started with this error 

**io.quarkus.builder.ChainBuildException: No producers for required item class io.quarkus.amazon.common.deployment.spi.EventLoopGroupBuildItem**

then i put these dependencies in the pom.xml to resolve, probabibly, the deployment missing class (the cognito one is needed for one error similar to previous one, but about other BuildItem class)

```xml
    <dependency>
      <groupId>io.quarkiverse.amazonservices</groupId>
      <artifactId>quarkus-amazon-common</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkiverse.amazonservices</groupId>
      <artifactId>quarkus-amazon-cognito-user-pools</artifactId>
    </dependency>
```

This solved the previous error but now i am getting this one

**Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: org.jboss.jandex.UnsupportedVersion: Can't read Jandex index from /META-INF/jandex.idx: Can't read index version 12; this IndexReader only supports index versions 2-3,6-11**

I tried to use version 3.0.x and 3.1.x of **jandex-maven-plugin** plugin, but the error persist.

Tried to exclude jandex from dependencies to 

```xml
      <exclusions>
        <exclusion>
          <groupId>io.smallrye</groupId>
          <artifactId>jandex</artifactId>
        </exclusion>
      </exclusions>
```

Someone can tell me what i am doing wrong?

**Comment by IncandescentChrysalis** [comment]
From [#40739](https://github.com/quarkusio/quarkus/issues/40739#issuecomment-2122639055), it seems you need to stick to a version <`3.2.0`, hence `3.1.8` should do the trick.

I face the same error on a project using quarkus over the Jandex Maven plugin `3.2.2`, and I solved it by setting the version to `3.1.8`:
```XML
<plugin>
    <groupId>io.smallrye</groupId>
    <artifactId>jandex-maven-plugin</artifactId>
    <version>${jandex.version}</version>
</plugin>
```

**Comment by gsmet** [comment]
I don’t know which version of Quarkus and Quarkus Amazon Services you are using but my guess is that the version of the latter is too new for the version of Quarkus you are using.
